src/tides.f90

```diff
@@ -573,4 +547,358 @@ subroutine setTidesInitFlag(this)
 
   end subroutine setTidesInitFlag
 
+!=========================================================================
+!
+!> @anchor      getDelaunay_arg
+!!
+!> @brief       Compute the Delaunay arguments (IERS chapter 5)
+!> @author      Andrea Turchi (ATU)
+!!
+!> @date        <ul>
+!!                <li> 17.08.22 (initial design) </li>
+!!              </ul>
+!!
+!!-----------------------------------------------------------------------
+  subroutine getDelaunay_arg(this, time_mjd, F_vect)
+
+    implicit none
+    class(Tides_class)                  :: this
+    real(dp), intent(in)                :: time_mjd                ! time in modified julian date
+    real(dp), parameter                 :: jd2000 = 2451545.0      ! factor to count julian days in jd2000 time reference
+    real(dp), parameter                 :: conv = 360.0/1296000.0  ! conversion factor from arcseconds to degrees 
+    real(dp)                            :: time_jd                 ! time in julian date
+    real(dp)                            :: jd_cent                 ! time in julian centuries
+    real(dp)                            :: l                       ! mean anomaly of the Moon
+    real(dp)                            :: l_prime                 ! mean anomaly of the Sun
+    real(dp)                            :: F                       ! mean longitude of the Moon minus Omega
+    real(dp)                            :: D                       ! mean elongation of the Moon from the Sun
+    real(dp)                            :: Omega                   ! mean longitude of the ascending node of the Moon
+    real(dp), dimension(5), intent(out) :: F_vect                  ! Delaunay arguments vector
+
+    !get input time in Julian centuries
+    time_jd = time_mjd + 2400000.5
+    jd_cent = (time_jd - jd2000)/36525.0
+    
+    !compute Delaunay arguments in degrees 
+    l = 134.96340251 + (1717915923.217800 * jd_cent + 31.879200 * jd_cent**2 &
+        + 0.05163500 * jd_cent**3 - 0.0002447000 * jd_cent**4) * conv
+    l_prime = 357.52910918 + (129596581.048100 * jd_cent - 0.553200 * jd_cent**2 &
+              + 0.00013600 * jd_cent**3 - 0.0000114900 * jd_cent**4) * conv
+    F = 93.27209062 + (1739527262.847800 * jd_cent - 12.751200 * jd_cent**2 &
+        - 0.00103700 * jd_cent**3 + 0.0000041700 * jd_cent**4) * conv
+    D = 297.85019547 + (1602961601.209000 * jd_cent - 6.370600 * jd_cent**2 &
+        + 0.00659300 * jd_cent**3 - 0.0000316900 * jd_cent**4) * conv
+    Omega = 125.04455501 + (- 6962890.543100 * jd_cent + 7.472200 * jd_cent**2 &
+            + 0.00770200 * jd_cent**3 - 0.0000593900 * jd_cent**4) * conv
```

Comment:
For constant values, use the "double" expression when used as double precision, e.g. 134.96340251d0 so that the compiler uses it as a double precision value